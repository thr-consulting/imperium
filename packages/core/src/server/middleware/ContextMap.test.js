import ContextMap from './ContextMap';

const myConnectors = {
	MyConnector: {},
};

function moduleFunc(/* connectors, context */) {
	return {
		MyDataModel: {
			fakeApiFn: () => 'fake api result',
		},
	};
}

describe('ContextMap', () => {
	it('should add modules properly', () => {
		const context = new ContextMap(myConnectors);
		context.addModule(moduleFunc);
		expect(context.getModel('MyDataModel')).toMatchSnapshot();
		expect(context.models.MyDataModel).toMatchSnapshot();
	});

	it('should set and get auth data', () => {
		const context = new ContextMap(myConnectors);
		context.auth = {name: 'test'};
		expect(context.auth).toMatchSnapshot();
	});

	it('should retrieve the connectors', () => {
		const context = new ContextMap(myConnectors);
		expect(context.connectors).toMatchSnapshot();
	});
});
