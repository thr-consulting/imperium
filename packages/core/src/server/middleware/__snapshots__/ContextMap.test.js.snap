// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMap should add modules properly 1`] = `
Object {
  "fakeApiFn": [Function],
}
`;

exports[`ContextMap should add modules properly 2`] = `
Object {
  "fakeApiFn": [Function],
}
`;

exports[`ContextMap should retrieve the connectors 1`] = `
Object {
  "MyConnector": Object {},
}
`;

exports[`ContextMap should set and get auth data 1`] = `
Object {
  "name": "test",
}
`;
